packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -0,0 +1,168 @@
+	 *
+	 * new PaginationFieldMessageEmbed().setTemplate(new MessageEmbed().setTitle('Test pager embed')).make().run(message.author, message.channel);
+	 * ```
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 * import { MessageEmbed } from 'discord.js';
+	 *
+	 * new PaginationFieldMessageEmbed().setTemplate({ title: 'Test pager embed' }).make().run(message.author, message.channel);
+	 * ```
+	 */
+	public setTemplate(template: MessageEmbedOptions | MessageEmbed) {
+		this.embedTemplate = template instanceof MessageEmbed ? template : new MessageEmbed(template);
+		return this;
+	}
+
+	/**
+	 * The way in which the items of the object array will be represented in the embed field to be paginated.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 * @param value Item information
+	 */
+	public formatField(value: (value: T, index: number, array: T[]) => any) {
+		this.items = this.items.map(value);
+		return this;
+	}
+
+	/**
+	 * Build the item pagination of the configured array.
+	 * It is required to put the `.make()` and `.run()` method at the end in order to start.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 */
+	public make() {
+		if (!this.fieldTitle.length) throw new Error('The title of the field to format must have a value.');
+		if (!this.items.length) throw new Error('The array introduced has no items.');
+		if (!this.items[0]) throw new Error('The format of the array items is incorrect.');
+
+		this.totalPages = Math.ceil(this.items.length / this.itemsPerPage);
+		this.makePages();
+		return this;
+	}
+
+	private makePages() {
+		for (let i = 0; i < this.totalPages; i++) {
+			const clonedTemplate = new MessageEmbed(
+				this.embedTemplate instanceof MessageEmbed ? (this.embedTemplate.toJSON() as MessageEmbedOptions) : this.embedTemplate
```

Comment:
Please move this check outside the loop, it's an expensive check that is always returning the same

Reply:
It is not possible to move it out of the loop because then no more clones would be created based on the embed template.

Reply:
I'm strictly talking about the ternary, not the constructor call!

Reply:
Done.